ai-ml/llm-finetuning-gemma/transformers/Dockerfile

```diff
@@ -0,0 +1,15 @@
+# [START gke_aiml_llm_finetune_gemma_single_node_docker]
+FROM nvidia/cuda:12.2.0-runtime-ubuntu22.04
+
+RUN apt-get update && \
+    apt-get -y --no-install-recommends install python3-dev gcc python3-pip git && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN pip3 install --no-cache-dir accelerate bitsandbytes datasets transformers peft trl torch
```

Comment:
In general it's preferable to check-in a requirements.txt file with pinned versions and let Renovate bump these versions, instead of doing it in the Dockerfile directly. The reason why is because with this method, the Dockerfile is not deterministic anymore; if I build this image today, vs. if I build this image in a few weeks, it'll create two completely different images (even if the content of the Dockerfile hasn't changed).

Reply:
Thats fair. Might just add versions of dependency in the Dockerfile. Otherwise its one more file for people to copy.

WDYT?